### PR TITLE
[mlir][tosa] Add fp16 support to `tosa.resize`

### PR DESCRIPTION
**Pull request by GeorgeARM**

### CODE REVIEW

**Comment by banach-space** [comment]
For those of us less familiar with the TOSA spec, could you provide some justification for supporting fp16 on top of fp32? Why not other FP types?

**Review by eric-k256 — APPROVED** [review]
Looks fine to me.

The only other floating point data type in TOSA is bf16, and those are unlikely to be used when doing a resize as resize is generally an image based operation where bf16 isn't common.

**Comment by GeorgeARM** [comment]
Thank you for your comment @banach-space.
As @eric-k256 mentioned the only remaining floating-point type supported in TOSA atm is bfloat16 and it is uncommon to have resize operations on it.
I would personally like to generalize the implementation to work on different floating-point types but this requires a more involved work. Happy to investigate this on a separate patch-set.

**Comment by banach-space** [comment]
> Thank you for your comment @banach-space. As @eric-k256 mentioned the only remaining floating-point type supported in TOSA atm is bfloat16 and it is uncommon to have resize operations on it. I would personally like to generalize the implementation to work on different floating-point types but this requires a more involved work. Happy to investigate this on a separate patch-set.

That is totally fine, but as somebody unfamiliar with the spec I'd appreciate a short note in the commit message :)

**Review by banach-space — APPROVED** [review]
LGTM :)

File: mlir/test/Conversion/TosaToLinalg/tosa-to-linalg-resize.mlir [hunk 1]
```diff
@@ -286,7 +304,7 @@ func.func @resize_bilinear_int(%arg0: tensor<1x19x20x1xi8>) {
 // -----
 
 // CHECK-LABEL: @resize_nearest_fp
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Conversion/TosaToLinalg/tosa-to-linalg-resize.mlir:306]: Please update this before landing in tree ;-)

**Comment by GeorgeARM** [comment]
> > Thank you for your comment @banach-space. As @eric-k256 mentioned the only remaining floating-point type supported in TOSA atm is bfloat16 and it is uncommon to have resize operations on it. I would personally like to generalize the implementation to work on different floating-point types but this requires a more involved work. Happy to investigate this on a separate patch-set.
> 
> That is totally fine, but as somebody unfamiliar with the spec I'd appreciate a short note in the commit message :)

Gotcha! Makes sense and apologies for any confusion!

**Pull request merged**